File: server/src/user/user.controller.ts
import { Controller, Get, Put, Body, Query, UseGuards } from '@nestjs/common';
import { UserService } from './user.service';
import { JwtAuthGuard } from '../auth/jwt-auth.guard';
import { CurrentUser } from '../auth/user.decorator';
import type { User } from '../auth/auth.service';

@Controller('users')
export class UserController {
	constructor(private readonly userService: UserService) { }

	@Get('search')
	search(@Query('q') q: string) {
		return this.userService.searchByEmail(q || '');
	}

	@UseGuards(JwtAuthGuard)
	@Put('profile')
	async updateProfile(@CurrentUser() user: User, @Body('name') name: string) {
		await this.userService.updateProfile(user.id, name);
		return { success: true };
	}

	@UseGuards(JwtAuthGuard)
	@Put('password')
	async updatePassword(@CurrentUser() user: User, @Body('password') password: string) {
		await this.userService.updatePassword(user.id, password);
		return { success: true };
	}
}

File: java-server/src/main/java/com/gistagent/user/UserController.java
package com.gistagent.user;

import java.util.List;
import java.util.Map;

import com.gistagent.auth.AuthenticatedUser;
import com.gistagent.user.dto.UpdatePasswordRequest;
import com.gistagent.user.dto.UpdateProfileRequest;
import com.gistagent.user.dto.UserSearchResult;

import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/users")
public class UserController {

	private final UserService userService;

	public UserController(UserService userService) {
		this.userService = userService;
	}

	@GetMapping("/search")
	public List<UserSearchResult> search(@RequestParam(name = "q", required = false) String q) {
		return userService.searchByEmail(q == null ? "" : q);
	}

	@PutMapping("/profile")
	public Map<String, Boolean> updateProfile(@AuthenticationPrincipal AuthenticatedUser user,
											  @RequestBody UpdateProfileRequest body) {
		userService.updateProfile(user.id(), body.name());
		return Map.of("success", true);
	}

	@PutMapping("/password")
	public Map<String, Boolean> updatePassword(@AuthenticationPrincipal AuthenticatedUser user,
											   @RequestBody UpdatePasswordRequest body) {
		userService.updatePassword(user.id(), body.password());
		return Map.of("success", true);
	}
}
